src/machine/machine_atsame5x_can.go

```diff
@@ -0,0 +1,456 @@
+// +build sam,atsame51 sam,atsame54
+
+package machine
+
+import (
+	"device/sam"
+	"errors"
+	"runtime/interrupt"
+	"unsafe"
+)
+
+const (
+	CANRxFifoSize = 16
+	CANTxFifoSize = 16
+	CANEvFifoSize = 16
+)
+
+// Message RAM can only be located in the first 64 KB area of the system RAM.
+// TODO: when the go:section pragma is merged, add the section configuration
+
+//go:align 4
+var CANRxFifo [2][(8 + 64) * CANRxFifoSize]byte
+
+//go:align 4
+var CANTxFifo [2][(8 + 64) * CANTxFifoSize]byte
+
+//go:align 4
+var CANEvFifo [2][(8) * CANEvFifoSize]byte
+
+type CAN struct {
+	Bus     *sam.CAN_Type
+	Channel byte
```

Comment:
It looks like `Channel` is unnecessary: it is always 0 when `Bus` equals `*sam.CAN0` and 1 when `Bus` equals `sam.CAN1`.
If you want to simplify the code, you could add this helper for example:

```go
func (can *CAN) instance() byte {
    if can.Bus == sam.CAN0 {
        return 0
    } else {
        return 1
    }
}
```

Reply:
Changed.